File: Latihan 2/PHP/Memory.php
<?php

/* Saya Silmi Aulia Rohmah mengerjakan Latihan 2 DPBO 2022 C1 
dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan 
kecurangan seperti yang telah dispesifikasikan. Aamiin */

include "Hardware.php"; //import class Hardware

class Memory extends Hardware{ //kelas Memory turunan dari kelas Hardware

    //atribut private kelas product
    private $frequency;
    private $MemorySize;
    private $supportsCuda;

    public function __construct(){ //konstruktor kosong

    }

    //metode getter dam setter
	public function getFrequency() {
		return $this->frequency;
	}

	public function setFrequency($frequency) {
		$this->frequency = $frequency;
	}

	public function getMemorySize() {
		return $this->MemorySize;
	}

	public function setMemorySize($MemorySize) {
		$this->MemorySize = $MemorySize;
	}

    	public function getSupportsCuda() {
		return $this->supportsCuda;
	}

	public function setSupportsCuda($supportsCuda) {
		$this->supportsCuda = $supportsCuda;
	}

    //untuk print atribut kelas Product
    public function getInfoMemory(){

        echo "Frequency     : ". $this->getFrequency()."<br/>";
        echo "Memory Size   : ". $this->getMemorySize()."<br/>";
        echo "Support Cude  : ". $this->getSupportsCuda()."<br/>";
    } 

	public function __destruct(){ //desktruktor

	}
}

File: Latihan 2/python/Memory.py
# Saya Silmi Aulia Rohmah mengerjakan Latihan 2 DPBO 2022 C1 
# dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan 
# kecurangan seperti yang telah dispesifikasikan. Aamiin 

from Hardware import Hardware

class Memory(Hardware): # kelas Memory turunan dari kelas Hardware

    def __init__(self): #Constructor

        # atribut
        self.frequency = "";
        self.MemorySize = "";
        self.supportsCuda = "";


    # metode getter dam setter
    def getFrequency(self):
        return self.frequency
    
    def setFrequency(self, frequency):
        self.frequency = frequency
    
    def getMemorySize(self):
        return self.MemorySize
        
    def setMemorySize(self, MemorySize):
        self.MemorySize = MemorySize

    def getSupportsCuda(self):
        return self.supportsCuda
        
    def setSupportsCuda(self, supportsCuda):
        self.supportsCuda = supportsCuda
      

    # untuk print atribut kelas Memory
    def getInfoMemory(self):
        print("Frequency     : " + self.getFrequency())
        print("Memory Size   : " + self.getMemorySize())
        print("Support Cude  : " + self.getSupportsCuda())
